lib/authApi.ts

```diff
@@ -0,0 +1,44 @@
+import { SignUpData } from "@/types/AuthTypes";
+import axios from "axios";
+
+const instance = axios.create({
+  baseURL: "https://panda-market-api.vercel.app/",
+});
+
+export const signUp = async (data: SignUpData) => {
+  try {
+    const response = await instance.post("/auth/signUp", data);
+
+    return response.data;
+  } catch (error) {
+    if (axios.isAxiosError(error)) {
+      console.error(
+        "가입 중 오류 발생:",
+        error.response ? error.response.data : error.message
+      );
+    } else {
+      console.error("가입 중 알 수 없는 오류 발생:", error);
+    }
+    throw error;
+  }
+};
+
+export const signIn = async ({ email, password }: SignUpData) => {
+  try {
+    const response = await instance.post("/auth/signIn", {
+      email,
+      password,
+    });
+
+    const { user, accessToken, refreshToken } = response.data;
+
+    // console.log("사용자:", user);
+    // console.log("액세스 토큰:", accessToken);
+    // console.log("리프레시 토큰:", refreshToken);
```

Comment:
콘솔은 PR 업로드 전 지우는 것을 추천드립니다. 😄

Reply:
다만, 현실적으로 통신을 통해 가져온 값들을 체크하면서 개발하면 편리한 부분이 있는 것은 사실입니다. 🤔 만약, 개발 환경에서 제한적으로 콘솔을 보고 싶은 니즈가 있다면 아래와 같이 작성할도 있습니다.
```suggestion
    if (process.env.NODE_ENV === 'development') {
      console.log("사용자:", user);
      console.log("액세스 토큰:", accessToken);
      console.log("리프레시 토큰:", refreshToken);
    }
```

Reply:
또한, 별도의 dev logger 의 역할을 하는 함수를 만든다면 조금 더 편하게 작성할 수 있고, 개발 로그 전체를 컨트롤 할 수 있어 조금 더 안전할 것 같네요. (다만, 실제 프로덕션 환경에서 자유롭게 쓰는 것은 고민해봐야 할 것 같습니다.)
```
const devLog = (...args: any[]): void => {
  if (process.env.NODE_ENV === 'development') {
    console.log(...args);
  }
};
```